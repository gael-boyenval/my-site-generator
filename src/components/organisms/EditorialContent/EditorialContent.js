import React from 'react';
import styled, { injectGlobal } from 'styled-components';
import PropTypes from 'prop-types';

import RehypeReact from 'rehype-react';

import { colors, unit } from 'Styles/tokens';
import { rem } from 'Styles/utils';
import prismTheme from 'Styles/prismTheme';

import Grid from 'Layout/Grid';
import Container from 'Layout/Container';
import { SubTitle, Title3Sans } from 'Atoms/Typography';

/* eslint-disable-next-line */
injectGlobal`${prismTheme}`;

const He2 = SubTitle.extend`
  margin: ${rem(unit * 2)} 0;
`;

const He3 = Title3Sans.extend`
  margin: ${rem(unit * 2)} 0;
`;

const TextContainer = styled.div`
  a {
    border-bottom: solid 1px currentColor;
  }
`;

const EditorialWrapper = styled.section`
  background: ${colors.white.light};
  color: ${colors.base.darkest};
`;

const renderAst = new RehypeReact({
  createElement: React.createElement,
  components: {
    he2: He2,
    he3: He3,
  },
}).Compiler;

const EditorialContent = ({ htmlAst }) => (
  <EditorialWrapper className="editorial-content">
    <Container>
      <Grid gutter>
        <Grid.Cell size={7 / 12} push={1 / 12}>
          <TextContainer>{renderAst(htmlAst)}</TextContainer>
        </Grid.Cell>
      </Grid>
    </Container>
  </EditorialWrapper>
);

EditorialContent.propTypes = {
  htmlAst: PropTypes.shape({}).isRequired,
};

export default EditorialContent;
