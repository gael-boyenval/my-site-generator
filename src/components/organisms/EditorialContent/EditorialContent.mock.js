export default {
  htmlAst: {
    type: 'root',
    children: [
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'div',
        properties: {},
        children: [
          {
            type: 'element',
            tagName: 'he2',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'Hello je suis un composant react',
              },
            ],
          },
          {
            type: 'text',
            value: '\n',
          },
          {
            type: 'element',
            tagName: 'he3',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'Hello je suis un composant react',
              },
            ],
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'text',
            value:
              'Do Pandas eat bananas? Check out this short video that shows that yes! pandas do seem to really enjoy bananas!',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'div',
        properties: {
          className: ['gatsby-highlight'],
        },
        children: [
          {
            type: 'text',
            value: '\n      ',
          },
          {
            type: 'element',
            tagName: 'pre',
            properties: {
              className: ['language-css'],
            },
            children: [
              {
                type: 'element',
                tagName: 'code',
                properties: {
                  className: ['language-css'],
                },
                children: [
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'selector'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: '.maclasse :',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' ',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: '{',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: '\n  ',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'property'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: 'color',
                      },
                    ],
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: ':',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' blue',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: ';',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: '\n',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: '}',
                      },
                    ],
                  },
                ],
              },
            ],
          },
          {
            type: 'text',
            value: '\n      ',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'div',
        properties: {
          className: ['gatsby-highlight'],
        },
        children: [
          {
            type: 'text',
            value: '\n      ',
          },
          {
            type: 'element',
            tagName: 'pre',
            properties: {
              className: ['language-javascript'],
            },
            children: [
              {
                type: 'element',
                tagName: 'code',
                properties: {
                  className: ['language-javascript'],
                },
                children: [
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'keyword'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: 'export',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' ',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'keyword'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: 'const',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' ',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'function-variable', 'function'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: 'closestLineHeight',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' ',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'operator'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: '=',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' ',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: '(',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: 'fontSize',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: ',',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' lineHeightRatio',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: ',',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' unit',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: ')',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' ',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'operator'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: '=>',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: '\n  ',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'function'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: 'closestMultiple',
                      },
                    ],
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: '(',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: 'fontSize ',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'operator'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: '*',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' lineHeightRatio',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: ',',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' unit',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: ')',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' ',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'operator'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: '/',
                      },
                    ],
                  },
                  {
                    type: 'text',
                    value: ' fontSize',
                  },
                  {
                    type: 'element',
                    tagName: 'span',
                    properties: {
                      className: ['token', 'punctuation'],
                    },
                    children: [
                      {
                        type: 'text',
                        value: ';',
                      },
                    ],
                  },
                ],
              },
            ],
          },
          {
            type: 'text',
            value: '\n      ',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'ul',
        properties: {},
        children: [
          {
            type: 'text',
            value: '\n',
          },
          {
            type: 'element',
            tagName: 'li',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'list item 1',
              },
            ],
          },
          {
            type: 'text',
            value: '\n',
          },
          {
            type: 'element',
            tagName: 'li',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'list item 2',
              },
            ],
          },
          {
            type: 'text',
            value: '\n',
          },
          {
            type: 'element',
            tagName: 'li',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'list item 3',
              },
            ],
          },
          {
            type: 'text',
            value: '\n',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'text',
            value: 'Emphasis, aka italics, with ',
          },
          {
            type: 'element',
            tagName: 'em',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'asterisks',
              },
            ],
          },
          {
            type: 'text',
            value: ' or ',
          },
          {
            type: 'element',
            tagName: 'em',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'underscores',
              },
            ],
          },
          {
            type: 'text',
            value: '.',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'text',
            value: 'Strong emphasis, aka bold, with ',
          },
          {
            type: 'element',
            tagName: 'strong',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'asterisks',
              },
            ],
          },
          {
            type: 'text',
            value: ' or ',
          },
          {
            type: 'element',
            tagName: 'strong',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'underscores',
              },
            ],
          },
          {
            type: 'text',
            value: '.',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'text',
            value: 'Combined emphasis with ',
          },
          {
            type: 'element',
            tagName: 'strong',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'asterisks and ',
              },
              {
                type: 'element',
                tagName: 'em',
                properties: {},
                children: [
                  {
                    type: 'text',
                    value: 'underscores',
                  },
                ],
              },
            ],
          },
          {
            type: 'text',
            value: '.',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'text',
            value: 'Strikethrough uses two tildes. ',
          },
          {
            type: 'element',
            tagName: 'del',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'Scratch this.',
              },
            ],
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'ol',
        properties: {},
        children: [
          {
            type: 'text',
            value: '\n',
          },
          {
            type: 'element',
            tagName: 'li',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'First ordered list item',
              },
            ],
          },
          {
            type: 'text',
            value: '\n',
          },
          {
            type: 'element',
            tagName: 'li',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'Another item\n⋅⋅* Unordered sub-list.',
              },
            ],
          },
          {
            type: 'text',
            value: '\n',
          },
          {
            type: 'element',
            tagName: 'li',
            properties: {},
            children: [
              {
                type: 'text',
                value:
                  "Actual numbers don't matter, just that it's a number\n⋅⋅1. Ordered sub-list",
              },
            ],
          },
          {
            type: 'text',
            value: '\n',
          },
          {
            type: 'element',
            tagName: 'li',
            properties: {},
            children: [
              {
                type: 'text',
                value: 'And another item.',
              },
            ],
          },
          {
            type: 'text',
            value: '\n',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'text',
            value:
              "⋅⋅⋅You can have properly indented paragraphs within list items. Notice the blank line above, and the leading spaces (at least one, but we'll use three here to also align the raw Markdown).",
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'text',
            value:
              '⋅⋅⋅To have a line break without a paragraph, you will need to use two trailing spaces.⋅⋅\n⋅⋅⋅Note that this line is separate, but within the same paragraph.⋅⋅\n⋅⋅⋅(This is contrary to the typical GFM line break behaviour, where trailing spaces are not required.)',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'ul',
        properties: {},
        children: [
          {
            type: 'text',
            value: '\n',
          },
          {
            type: 'element',
            tagName: 'li',
            properties: {},
            children: [
              {
                type: 'text',
                value: '\n',
              },
              {
                type: 'element',
                tagName: 'p',
                properties: {},
                children: [
                  {
                    type: 'text',
                    value: 'Unordered list can use asterisks',
                  },
                ],
              },
              {
                type: 'text',
                value: '\n',
              },
            ],
          },
          {
            type: 'text',
            value: '\n',
          },
          {
            type: 'element',
            tagName: 'li',
            properties: {},
            children: [
              {
                type: 'text',
                value: '\n',
              },
              {
                type: 'element',
                tagName: 'p',
                properties: {},
                children: [
                  {
                    type: 'text',
                    value: 'Or minuses',
                  },
                ],
              },
              {
                type: 'text',
                value: '\n',
              },
            ],
          },
          {
            type: 'text',
            value: '\n',
          },
          {
            type: 'element',
            tagName: 'li',
            properties: {},
            children: [
              {
                type: 'text',
                value: '\n',
              },
              {
                type: 'element',
                tagName: 'p',
                properties: {},
                children: [
                  {
                    type: 'text',
                    value: 'Or pluses',
                  },
                ],
              },
              {
                type: 'text',
                value: '\n',
              },
            ],
          },
          {
            type: 'text',
            value: '\n',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'element',
            tagName: 'a',
            properties: {
              href: 'https://www.google.com',
            },
            children: [
              {
                type: 'text',
                value: "I'm an inline-style link",
              },
            ],
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'element',
            tagName: 'a',
            properties: {
              href: 'https://www.google.com',
              title: "Google's Homepage",
            },
            children: [
              {
                type: 'text',
                value: "I'm an inline-style link with title",
              },
            ],
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'element',
            tagName: 'a',
            properties: {
              href: 'https://www.mozilla.org',
            },
            children: [
              {
                type: 'text',
                value: "I'm a reference-style link",
              },
            ],
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'element',
            tagName: 'a',
            properties: {
              href: '../blob/master/LICENSE',
            },
            children: [
              {
                type: 'text',
                value: "I'm a relative reference to a repository file",
              },
            ],
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'element',
            tagName: 'a',
            properties: {
              href: 'http://slashdot.org',
            },
            children: [
              {
                type: 'text',
                value: 'You can use numbers for reference-style link definitions',
              },
            ],
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'text',
            value: 'Or leave it empty and use the ',
          },
          {
            type: 'element',
            tagName: 'a',
            properties: {
              href: 'http://www.reddit.com',
            },
            children: [
              {
                type: 'text',
                value: 'link text itself',
              },
            ],
          },
          {
            type: 'text',
            value: '.',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'text',
            value: 'URLs and URLs in angle brackets will automatically get turned into links.\n',
          },
          {
            type: 'element',
            tagName: 'a',
            properties: {
              href: 'http://www.example.com',
            },
            children: [
              {
                type: 'text',
                value: 'http://www.example.com',
              },
            ],
          },
          {
            type: 'text',
            value: ' or ',
          },
          {
            type: 'element',
            tagName: 'a',
            properties: {
              href: 'http://www.example.com',
            },
            children: [
              {
                type: 'text',
                value: 'http://www.example.com',
              },
            ],
          },
          {
            type: 'text',
            value: ' and sometimes\nexample.com (but not on Github, for example).',
          },
        ],
      },
      {
        type: 'text',
        value: '\n',
      },
      {
        type: 'element',
        tagName: 'p',
        properties: {},
        children: [
          {
            type: 'text',
            value: 'Some text to show that the reference links can follow later.',
          },
        ],
      },
    ],
    data: {
      quirksMode: false,
    },
  },
};
