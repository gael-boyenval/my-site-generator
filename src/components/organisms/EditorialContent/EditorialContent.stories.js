/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs';

import EditorialContent from './EditorialContent';
import propMock from './EditorialContent.mock';

storiesOf('04 - Organisms', module)
  .addDecorator(withKnobs)
  .add('EditorialContent', withInfo()(() => <EditorialContent {...propMock} />));
