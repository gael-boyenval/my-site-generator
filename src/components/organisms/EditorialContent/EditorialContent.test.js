/* eslint-disable no-undef */
import React from 'react';

import EditorialContent from './EditorialContent';
import propMock from './EditorialContent.mock';

describe('Component EditorialContent', () => {
  it('should be render without crash', () => {
    const component = shallow(<EditorialContent {...propMock} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
