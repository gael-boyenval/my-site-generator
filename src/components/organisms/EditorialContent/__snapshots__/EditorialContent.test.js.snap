// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component EditorialContent should be render without crash 1`] = `
<styled.section
  className="editorial-content"
>
  <styled.div>
    <styled.div
      gutter={true}
    >
      <styled.div
        push={0.08333333333333333}
        size={0.5833333333333334}
      >
        <styled.div>
          <div
            key="h-1"
          >
            
            
            <div
              key="h-2"
            >
              <styled.h2
                key="h-3"
              >
                Hello je suis un composant react
              </styled.h2>
              
              
              <styled.h3
                key="h-4"
              >
                Hello je suis un composant react
              </styled.h3>
            </div>
            
            
            <p
              key="h-5"
            >
              Do Pandas eat bananas? Check out this short video that shows that yes! pandas do seem to really enjoy bananas!
            </p>
            
            
            <div
              className="gatsby-highlight"
              key="h-6"
            >
              
                    
              <pre
                className="language-css"
                key="h-7"
              >
                <code
                  className="language-css"
                  key="h-8"
                >
                  <span
                    className="token selector"
                    key="h-9"
                  >
                    .maclasse :
                  </span>
                   
                  <span
                    className="token punctuation"
                    key="h-10"
                  >
                    {
                  </span>
                  
                    
                  <span
                    className="token property"
                    key="h-11"
                  >
                    color
                  </span>
                  <span
                    className="token punctuation"
                    key="h-12"
                  >
                    :
                  </span>
                   blue
                  <span
                    className="token punctuation"
                    key="h-13"
                  >
                    ;
                  </span>
                  
                  
                  <span
                    className="token punctuation"
                    key="h-14"
                  >
                    }
                  </span>
                </code>
              </pre>
              
                    
            </div>
            
            
            <div
              className="gatsby-highlight"
              key="h-15"
            >
              
                    
              <pre
                className="language-javascript"
                key="h-16"
              >
                <code
                  className="language-javascript"
                  key="h-17"
                >
                  <span
                    className="token keyword"
                    key="h-18"
                  >
                    export
                  </span>
                   
                  <span
                    className="token keyword"
                    key="h-19"
                  >
                    const
                  </span>
                   
                  <span
                    className="token function-variable function"
                    key="h-20"
                  >
                    closestLineHeight
                  </span>
                   
                  <span
                    className="token operator"
                    key="h-21"
                  >
                    =
                  </span>
                   
                  <span
                    className="token punctuation"
                    key="h-22"
                  >
                    (
                  </span>
                  fontSize
                  <span
                    className="token punctuation"
                    key="h-23"
                  >
                    ,
                  </span>
                   lineHeightRatio
                  <span
                    className="token punctuation"
                    key="h-24"
                  >
                    ,
                  </span>
                   unit
                  <span
                    className="token punctuation"
                    key="h-25"
                  >
                    )
                  </span>
                   
                  <span
                    className="token operator"
                    key="h-26"
                  >
                    =&gt;
                  </span>
                  
                    
                  <span
                    className="token function"
                    key="h-27"
                  >
                    closestMultiple
                  </span>
                  <span
                    className="token punctuation"
                    key="h-28"
                  >
                    (
                  </span>
                  fontSize 
                  <span
                    className="token operator"
                    key="h-29"
                  >
                    *
                  </span>
                   lineHeightRatio
                  <span
                    className="token punctuation"
                    key="h-30"
                  >
                    ,
                  </span>
                   unit
                  <span
                    className="token punctuation"
                    key="h-31"
                  >
                    )
                  </span>
                   
                  <span
                    className="token operator"
                    key="h-32"
                  >
                    /
                  </span>
                   fontSize
                  <span
                    className="token punctuation"
                    key="h-33"
                  >
                    ;
                  </span>
                </code>
              </pre>
              
                    
            </div>
            
            
            <ul
              key="h-34"
            >
              
              
              <li
                key="h-35"
              >
                list item 1
              </li>
              
              
              <li
                key="h-36"
              >
                list item 2
              </li>
              
              
              <li
                key="h-37"
              >
                list item 3
              </li>
              
              
            </ul>
            
            
            <p
              key="h-38"
            >
              Emphasis, aka italics, with 
              <em
                key="h-39"
              >
                asterisks
              </em>
               or 
              <em
                key="h-40"
              >
                underscores
              </em>
              .
            </p>
            
            
            <p
              key="h-41"
            >
              Strong emphasis, aka bold, with 
              <strong
                key="h-42"
              >
                asterisks
              </strong>
               or 
              <strong
                key="h-43"
              >
                underscores
              </strong>
              .
            </p>
            
            
            <p
              key="h-44"
            >
              Combined emphasis with 
              <strong
                key="h-45"
              >
                asterisks and 
                <em
                  key="h-46"
                >
                  underscores
                </em>
              </strong>
              .
            </p>
            
            
            <p
              key="h-47"
            >
              Strikethrough uses two tildes. 
              <del
                key="h-48"
              >
                Scratch this.
              </del>
            </p>
            
            
            <ol
              key="h-49"
            >
              
              
              <li
                key="h-50"
              >
                First ordered list item
              </li>
              
              
              <li
                key="h-51"
              >
                Another item
                ⋅⋅* Unordered sub-list.
              </li>
              
              
              <li
                key="h-52"
              >
                Actual numbers don't matter, just that it's a number
                ⋅⋅1. Ordered sub-list
              </li>
              
              
              <li
                key="h-53"
              >
                And another item.
              </li>
              
              
            </ol>
            
            
            <p
              key="h-54"
            >
              ⋅⋅⋅You can have properly indented paragraphs within list items. Notice the blank line above, and the leading spaces (at least one, but we'll use three here to also align the raw Markdown).
            </p>
            
            
            <p
              key="h-55"
            >
              ⋅⋅⋅To have a line break without a paragraph, you will need to use two trailing spaces.⋅⋅
              ⋅⋅⋅Note that this line is separate, but within the same paragraph.⋅⋅
              ⋅⋅⋅(This is contrary to the typical GFM line break behaviour, where trailing spaces are not required.)
            </p>
            
            
            <ul
              key="h-56"
            >
              
              
              <li
                key="h-57"
              >
                
                
                <p
                  key="h-58"
                >
                  Unordered list can use asterisks
                </p>
                
                
              </li>
              
              
              <li
                key="h-59"
              >
                
                
                <p
                  key="h-60"
                >
                  Or minuses
                </p>
                
                
              </li>
              
              
              <li
                key="h-61"
              >
                
                
                <p
                  key="h-62"
                >
                  Or pluses
                </p>
                
                
              </li>
              
              
            </ul>
            
            
            <p
              key="h-63"
            >
              <a
                href="https://www.google.com"
                key="h-64"
              >
                I'm an inline-style link
              </a>
            </p>
            
            
            <p
              key="h-65"
            >
              <a
                href="https://www.google.com"
                key="h-66"
                title="Google's Homepage"
              >
                I'm an inline-style link with title
              </a>
            </p>
            
            
            <p
              key="h-67"
            >
              <a
                href="https://www.mozilla.org"
                key="h-68"
              >
                I'm a reference-style link
              </a>
            </p>
            
            
            <p
              key="h-69"
            >
              <a
                href="../blob/master/LICENSE"
                key="h-70"
              >
                I'm a relative reference to a repository file
              </a>
            </p>
            
            
            <p
              key="h-71"
            >
              <a
                href="http://slashdot.org"
                key="h-72"
              >
                You can use numbers for reference-style link definitions
              </a>
            </p>
            
            
            <p
              key="h-73"
            >
              Or leave it empty and use the 
              <a
                href="http://www.reddit.com"
                key="h-74"
              >
                link text itself
              </a>
              .
            </p>
            
            
            <p
              key="h-75"
            >
              URLs and URLs in angle brackets will automatically get turned into links.
              
              <a
                href="http://www.example.com"
                key="h-76"
              >
                http://www.example.com
              </a>
               or 
              <a
                href="http://www.example.com"
                key="h-77"
              >
                http://www.example.com
              </a>
               and sometimes
              example.com (but not on Github, for example).
            </p>
            
            
            <p
              key="h-78"
            >
              Some text to show that the reference links can follow later.
            </p>
          </div>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.section>
`;
